lib/web_ui/lib/src/engine/semantics/text_field.dart

```diff
@@ -341,75 +378,109 @@ class TextField extends RoleManager {
     }), true);
   }
 
-  bool _hasFocused = false;
-
-  @override
-  void update() {
-    // The user is editing the semantic text field directly, so there's no need
-    // to do any update here.
-    if (semanticsObject.hasLabel) {
-      editableElement.setAttribute(
-        'aria-label',
-        semanticsObject.label!,
-      );
-    } else {
-      editableElement.removeAttribute('aria-label');
-    }
-
-    editableElement.style
-      ..width = '${semanticsObject.rect!.width}px'
-      ..height = '${semanticsObject.rect!.height}px';
+  void _applyTextFromEditingState() {
+    final EditingState editingState = EditingState(
+      text: semanticsObject.value,
+      baseOffset: semanticsObject.textSelectionBase,
+      extentOffset: semanticsObject.textSelectionExtent,
+    );
+    editingState.applyTextToDomElement(editableElement);
+  }
 
-    // Whether we should request that the browser shift focus to the editable
-    // element, so that both the framework and the browser agree on what's
-    // currently focused.
-    bool needsDomFocusRequest = false;
+  void _setEditingState() {
     final EditingState editingState = EditingState(
       text: semanticsObject.value,
       baseOffset: semanticsObject.textSelectionBase,
       extentOffset: semanticsObject.textSelectionExtent,
     );
-    if (semanticsObject.hasFocus) {
-      if (!_hasFocused) {
-        _hasFocused = true;
-        SemanticsTextEditingStrategy.instance.activate(this);
-        needsDomFocusRequest = true;
-      }
-      if (domDocument.activeElement != editableElement) {
-        needsDomFocusRequest = true;
-      }
-      // Focused elements should have full text editing state applied.
-      SemanticsTextEditingStrategy.instance.setEditingState(editingState);
-    } else if (_hasFocused) {
+    SemanticsTextEditingStrategy.instance.setEditingState(editingState);
+  }
+
+  void _invokeIosWorkaround() {
+    if (editableElement != null) {
+      return;
+    }
+
+    _initializeEditableElement();
+    activeEditableElement.style.transform = 'translate(-999px,-999px)';
```

Comment:
With 8k screen resolutions getting to 7680x4320 these days, I wonder if 999 is a sufficient  #offset. Would using a bigger number, e.g. 20000, work?

Reply:
You're right. I've increased it to 9999 to match what we have in the non semantic iOS strategy.